asllib/StaticOperations.ml

```diff
@@ -310,32 +310,45 @@ module Make (C : CONFIG) = struct
     in
     function [] -> [] | x :: li -> aux x [ x ] li
 
+  let dz_unions =
+    let module DZ = Diet.Z in
+    let rec main = function
+      | [] -> DZ.empty
+      | [ x ] -> x
+      | li -> aux [] li |> main
+    and aux acc = function
+      | [] -> acc
+      | [ x ] -> x :: acc
+      | x :: y :: li -> aux (DZ.union x y :: acc) li
+    in
+    main
+
   let simplify_static_constraints =
     let module DZ = Diet.Z in
-    let acc_diet (diet, non_static) = function
+    let to_diets (diets, non_static) = function
       | Constraint_Exact e as c -> (
           match e.desc with
-          | E_Literal (L_Int z) ->
-              (DZ.(add Interval.(make z z) diet), non_static)
-          | _ -> (diet, c :: non_static))
+          | E_Literal (L_Int z) -> (DZ.singleton z :: diets, non_static)
+          | _ -> (diets, c :: non_static))
       | Constraint_Range (e1, e2) as c -> (
           match (e1.desc, e2.desc) with
           | E_Literal (L_Int z1), E_Literal (L_Int z2) when Z.leq z1 z2 ->
-              DZ.(add Interval.(make z1 z2) diet, non_static)
-          | _ -> (diet, c :: non_static))
+              DZ.(add Interval.(make z1 z2) empty :: diets, non_static)
+          | _ -> (diets, c :: non_static))
     in
     let constraint_of_interval interval =
       let x = DZ.Interval.x interval and y = DZ.Interval.y interval in
       if Z.equal x y then Constraint_Exact (expr_of_z x)
       else Constraint_Range (expr_of_z x, expr_of_z y)
     in
     fun constraints ->
-      let diet, non_static =
-        List.fold_left acc_diet (DZ.empty, []) constraints
-      in
-      DZ.fold
-        (fun interval acc -> constraint_of_interval interval :: acc)
-        diet non_static
+      if List.length constraints > 1000 then constraints
```

Comment:
Not sure I follow what this check aims to achieve - would this limit also need documenting?